crates/bevy_reflect/bevy_reflect_derive/src/container_attributes.rs

```diff
@@ -23,19 +24,38 @@ const HASH_ATTR: &str = "Hash";
 // but useful to know exist nonetheless
 pub(crate) const REFLECT_DEFAULT: &str = "ReflectDefault";
 
+// The error message to show when a trait is specified multiple times
+const CONFLICTING_IMPL_MESSAGE: &str = "conflicting reflected trait implementation";
+
 /// A marker for trait implementations registered via the `Reflect` derive macro.
 #[derive(Clone, Default)]
 pub(crate) enum TraitImpl {
     /// The trait is not registered as implemented.
     #[default]
     NotImplemented,
+
     /// The trait is registered as implemented.
-    Implemented,
+    Implemented(Span),
 
     // TODO: This can be made to use `ExprPath` instead of `Ident`, allowing for fully qualified paths to be used
     /// The trait is registered with a custom function rather than an actual implementation.
-    Custom(Ident),
+    Custom(Ident, Span),
```

Comment:
I wonder if, while we're here, we could implement this TODO. I imagine using `ExprPath` will affect the way we handle this `Span`. Thoughts on this? Should we do it in this PR or save it for a separate one?

Reply:
I can give it a shot in this PR. I think the change is pretty straightforward (just replace `Ident` with `Path` or `ExprPath`). The `Span` is associated with the whole registration item (the `Hash(my_hash)` part of `#[reflect(Hash(my_hash))]`) while the `Ident`/`ExprPath` is associated with just the `my_hash` part of that.

Reply:
Looks like `NestedMeta` doesn't provide an `ExprPath`, but it does provide a `Path` so I'll use that. The difference is that `ExprPath` supports things like `<Self as Trait>::foo<i32>` whereas `Path` only supports things like `Trait::foo<i32>` I believe.